src/Raven.Server/Documents/Sharding/Comparers/DocumentsComparer.cs

```diff
@@ -10,31 +10,53 @@
 
 namespace Raven.Server.Documents.Sharding.Comparers;
 
-public class DocumentsComparer : IComparer<BlittableJsonReaderObject>
+public sealed class DocumentsComparer : IComparer<BlittableJsonReaderObject>
 {
     private readonly OrderByField[] _orderByFields;
     private readonly bool _extractFromData;
+    private readonly Random[] _randoms;
 
-    public DocumentsComparer(OrderByField[] orderByFields, bool extractFromData)
+    public DocumentsComparer(OrderByField[] orderByFields, bool extractFromData, bool hasOrderByRandom)
     {
         _orderByFields = orderByFields;
         _extractFromData = extractFromData;
+        _randoms = hasOrderByRandom == false
+            ? null
+            : GetRandom(_orderByFields);
     }
+    
+    private Random[] GetRandom(OrderByField[] orderByFields)
+    {
+        var randoms = new Random[orderByFields.Length];
+        
+        for (int idX = 0, randomIdx = 0; idX < _orderByFields.Length; ++idX)
+        {
+            if (orderByFields[idX].OrderingType != OrderByFieldType.Random)
+                continue;
+
+            ref var randomField = ref orderByFields[idX];
+            randoms[randomIdx++] = randomField.Arguments is {Length: > 0} 
+                ? new Random(randomField.Arguments[0].NameOrValue.GetHashCode()) 
```

Comment:
IIRC `GetHashCode()` might be not stable across different process

Reply:
Good catch :) 
Changed to `xxHash` from Sparrow.